scapy/contrib/automotive/someip.py

```diff
@@ -0,0 +1,151 @@
+#! /usr/bin/env python
+
+# MIT License
+
+# Copyright (c) 2018 Jose Amores
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more information
+# Copyright (C) Sebastian Baar <sebastian.baar@gmx.de>
+# This program is published under a GPLv2 license
+
+from scapy.layers.inet import TCP, UDP
+from scapy.all import struct, bind_layers
```

Comment:
Nope. Never import from `scapy.all`, particularly built-in Python modules!

Add `import struct` before the Scapy imports. And import `bind_layers` from `scapy.packet`.

Also, import `range` from `scapy.modules.six.moves` since you use it.

Reply:
`range` is still a built-in in Python 3, but it’s a generator whereas it was a list on Python 2.

It’s sort of unwanted to have all those `import range` in scapy, but it was done automatically during the `six` setup. Most of them have already been removed since them (same issue with `map`)

Reply:
It is **not** unwanted. We would write `for i in xrange()` in Python 2 and `for i in range()`, so we import `range` to have the same thing and avoid a useless list creation in Python 2.

Reply:
Done.